arbos/l2pricing/model.go

```diff
@@ -22,16 +22,37 @@ const InitialRateEstimateInertia = 60
 var InitialGasPoolTargetBips = arbmath.PercentToBips(80)
 var InitialGasPoolWeightBips = arbmath.PercentToBips(60)
 
-func (ps *L2PricingState) AddToGasPool(gas int64) error {
+const FirstExponentialPricingVersion = 4
+
+func (ps *L2PricingState) AddToGasPool(gas int64, arbosVersion uint64) error {
 	gasPool, err := ps.GasPool()
 	if err != nil {
 		return err
 	}
+	if arbosVersion >= FirstExponentialPricingVersion && gasPool >= 0 {
+		gasPool = 0
+	}
 	return ps.SetGasPool(arbmath.SaturatingAdd(gasPool, gas))
 }
 
 // Update the pricing model with info from the last block
 func (ps *L2PricingState) UpdatePricingModel(l2BaseFee *big.Int, timePassed uint64, arbosVersion uint64, debug bool) {
+	if arbosVersion < FirstExponentialPricingVersion {
+		// note: if we restart the chain at version >= FirstExponentialPricingVersion, we can simplify the L2-pricing-related params and precompiles
+		ps.UpdatePricingModelPreExpVersion(l2BaseFee, timePassed, arbosVersion, debug)
+		return
+	}
+
+	speedLimit, _ := ps.SpeedLimitPerSecond()
+	_ = ps.AddToGasPool(int64(timePassed*speedLimit), arbosVersion)
+	gasPool, _ := ps.GasPool()
+	minBaseFee, _ := ps.MinBaseFeeWei()
+	expVal := arbmath.ApproxExpBasisPoints(arbmath.Bips(-arbmath.SaturatingMul(gasPool, int64(arbmath.OneInBips)) / (int64(102 * speedLimit))))
```

Comment:
While there's theory backing the current value for `B0`, we should probably still make it adjustable via an `ArbOwner` precompile method.

Reply:
Added a parameter for this, and one for a tolerance (an amount of backlog that can occur before price starts escalating).